nexus-repository-r/src/main/java/org/sonatype/nexus/repository/r/internal/RPackagesInformationBuilder.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2017-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.r.internal;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.util.Enumeration;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.TreeMap;
+
+import javax.mail.internet.InternetHeaders;
+
+import org.sonatype.nexus.repository.storage.Asset;
+
+import org.apache.commons.compress.compressors.CompressorException;
+import org.apache.commons.compress.compressors.CompressorOutputStream;
+import org.apache.commons.compress.compressors.CompressorStreamFactory;
+
+import static com.google.common.base.Charsets.UTF_8;
+import static java.util.Collections.unmodifiableMap;
+import static org.apache.commons.compress.compressors.CompressorStreamFactory.GZIP;
+import static org.sonatype.nexus.repository.r.internal.RAttributes.P_DEPENDS;
+import static org.sonatype.nexus.repository.r.internal.RAttributes.P_IMPORTS;
+import static org.sonatype.nexus.repository.r.internal.RAttributes.P_LICENSE;
+import static org.sonatype.nexus.repository.r.internal.RAttributes.P_NEEDS_COMPILATION;
+import static org.sonatype.nexus.repository.r.internal.RAttributes.P_PACKAGE;
+import static org.sonatype.nexus.repository.r.internal.RAttributes.P_SUGGESTS;
+import static org.sonatype.nexus.repository.r.internal.RAttributes.P_VERSION;
+
+/**
+ * Builds the contents of a PACKAGES.gz file based on the provided assets, taking into account the greatest version of a
+ * particular package that is available in a (hosted) repository.
+ *
+ * Note that this maintains all pertinent information for the "latest" version of each package in memory, though the
+ * actual amount of information for each package is rather small.
+ *
+ * TODO: Add support for other metadata types (PACKAGES, PACKAGES.rds etc.)
+ */
+public class RPackagesInformationBuilder
```

Comment:
Please add 'since 1.next' value

Reply:
done in fe40919080a00ea9162d0af9be02c414af338e27